EasyLayout.Forms/EasyLayout.cs

```diff
@@ -682,66 +603,103 @@ private static Position GetPosition(MethodCallExpression fExpr)
-                case nameof(Rectangle.Top):
+                case nameof(Y):
                     return Position.Top;
-                case nameof(Rectangle.Right):
+                case nameof(Left):
+                    return Position.Left;
+                case nameof(X):
+                    return Position.Left;
+                case nameof(Right):
                     return Position.Right;
-                case nameof(Rectangle.Bottom):
+                case nameof(Bottom):
                     return Position.Bottom;
-                case nameof(Rectangle.Width):
-                    return Position.Width;
-                case nameof(Rectangle.Height):
+                case nameof(Height):
                     return Position.Height;
-                case nameof(Rectangle.Center):
-                    return Position.Center;
+                case nameof(Width):
+                    return Position.Width;
+                case nameof(CenterX):
+                    return Position.CenterX;
+                case nameof(CenterY):
+                    return Position.CenterY;
                 default:
-                    throw new NotSupportedException("Property " + memExpr.Member.Name + " is not recognized.");
+                    throw new NotSupportedException("Method " + fExpr.Method.Name + " is not recognized.");
             }
         }
 
-        private static List<BinaryExpression> FindConstraints(Expression expr)
+        private static IEnumerable<BinaryExpression> FindConstraints(Expression expr)
         {
             var binaryExpressions = new List<BinaryExpression>();
             FindConstraints(expr, binaryExpressions);
             return binaryExpressions;
         }
 
-        private static void FindConstraints(Expression expr, List<BinaryExpression> constraintExprs)
+        private static void FindConstraints(Expression expr, ICollection<BinaryExpression> constraintExprs)
         {
-            var b = expr as BinaryExpression;
-            if (b == null)
+            if (expr is BinaryExpression bExpr)
             {
-                return;
+                if (bExpr.NodeType == ExpressionType.AndAlso)
+                {
+                    FindConstraints(bExpr.Left, constraintExprs);
+                    FindConstraints(bExpr.Right, constraintExprs);
+                }
+                else
+                {
+                    constraintExprs.Add(bExpr);
+                }
             }
+        }
 
-            if (b.NodeType == ExpressionType.AndAlso)
-            {
-                FindConstraints(b.Left, constraintExprs);
-                FindConstraints(b.Right, constraintExprs);
-            }
-            else
-            {
-                constraintExprs.Add(b);
-            }
+        // Helper extension methods to make it easier to do the relative layout
+        public static int Top(this VisualElement visualElement)
+        {
+            return 0;
```

Comment:
I like the idea of throwing the exception, it provides a clear hint as to why the methods are there!